src/views/Users/ViewUser/index.jsx

```diff
@@ -0,0 +1,470 @@
+              onChange={e =>
+                handlePermissionNameChange(permission, index, e.target.value)
+              }
+            />
+          )}
+        </Grid>
+        {row[1] === undefined && <Grid item xs />}
+        {row[1] !== undefined && row[1] !== 'add' && (
+          <Fragment>
+            <Grid item xs>
+              <TextField
+                value={row[1]}
+                style={{ width: '85%' }}
+                onChange={e =>
+                  handleProductNameChange(permission, index, e.target.value)
+                }
+              />
+              <IconButton
+                onClick={() => handleProductDelete(permission, index)}
+                style={{ width: '15%' }}>
+                <DeleteIcon />
+              </IconButton>
+            </Grid>
+          </Fragment>
+        )}
+        {row[1] === 'add' && (
+          <Grid item xs className={classes.addGrid}>
+            <Button
+              onClick={() => handleProductAdd(permission)}
+              className={classes.fullWidth}
+              variant="outlined">
+              <PlusIcon />
+            </Button>
+          </Grid>
+        )}
+        {row[2] === undefined && <Grid item xs />}
+        {row[2] !== undefined && row[2] !== 'add' && (
+          <Fragment>
+            <Grid item xs>
+              <TextField
+                value={row[2]}
+                onChange={e =>
+                  handleActionNameChange(permission, index, e.target.value)
+                }
+                style={{ width: '85%' }}
+              />
+              <IconButton
+                onClick={() => handleActionDelete(permission, index)}
+                style={{ width: '15%' }}>
+                <DeleteIcon />
+              </IconButton>
+            </Grid>
+          </Fragment>
+        )}
+        {row[2] === 'add' && (
+          <Grid item xs className={classes.addGrid}>
+            <Button
+              className={classes.fullWidth}
+              variant="outlined"
+              onClick={() => handleActionAdd(permission)}>
+              <PlusIcon />
+            </Button>
+          </Grid>
+        )}
+      </Grid>
+    ));
+
+  return (
+    <Dashboard title="Users">
+      {error && <ErrorPanel error={error} />}
+      {isLoading && <Spinner loading />}
+      {!isLoading && (
+        <Fragment>
+          <form autoComplete="off">
+            <Grid container spacing={2}>
+              <Grid item xs>
+                <TextField
+                  required
+                  disabled={!isNewUser}
+                  onChange={handleUsernameChange}
+                  fullWidth
+                  label="Username"
+                  value={username}
+                />
+              </Grid>
+            </Grid>
+            <br />
+            <br />
+            <br />
+            <Typography variant="h5">Roles</Typography>
+            {roles.map(renderRole)}
```

Comment:
Let's make the text field occupy the full width of the available space in the main window:

#### Before

![Screen Shot 2019-07-04 at 10 22 09 AM](https://user-images.githubusercontent.com/3766511/60673402-a0402080-9e45-11e9-88da-604b361f8473.png)

#### After

![Screen Shot 2019-07-04 at 10 17 08 AM](https://user-images.githubusercontent.com/3766511/60673112-037d8300-9e45-11e9-9719-76d28aa444d4.png)

We would need to add the `fullWidth` prop to the `TextField` component, then modify the first grid item to have `xs={11}` and then the second one to `xs={1}`. In order to have the garbage icon to the far right, we would need to add a class name to the second column. I think something like:

```jsx
gridRoleDelete: {
    display: 'flex',
    justifyContent: 'flex-end',
},

// ...

<Grid item xs={11} ... />
<Grid item xs={1} className={classes.gridRoleDelete}  ... />

```